src/io/read_tree_genesis_hdf5.c

```diff
@@ -423,12 +427,27 @@ int setup_forests_io_genesis_hdf5(struct forest_info *forests_info, const int Th
             char dset_name[MAX_STRING_LEN + 1];
             snprintf(dset_name, MAX_STRING_LEN, "Snap_%03d", isnap);
             const hsize_t h5_single_item = 1;
-            READ_GENESIS_PARTIAL_FORESTINFO(h5_fd, dset_name, "ForestOffsetPerSnap", &offset_for_forestnum, &h5_single_item, &snap_offset);
+            const hsize_t ndim = 1;
+            READ_GENESIS_PARTIAL_FORESTINFO(h5_fd, dset_name, "ForestOffsetPerSnap", ndim, &offset_for_forestnum, &h5_single_item, &snap_offset);
             gen->halo_offset_per_snap[isnap] = snap_offset;
             /* fprintf(stderr,"On ThisTask = %d done with calculating halo offsets for first file = '%s' at snap = %d start_forestnum_per_file = %"PRId64"\n", */
             /*         ThisTask, fname, isnap, start_forestnum_per_file[start_filenum]); */
             run_params->interrupted=1;
         }
+#else
+#warning Using 'ForestInfoInFile/ForestOffsetsAllSnaps'
+        const hsize_t start_forestnum_in_file = forests_info->original_treenr[0];
+        const hsize_t read_ndim = 2;
+        const hsize_t read_offset[2] = {start_forestnum_in_file, 0};
+        const hsize_t read_count[2] = {1, gen->maxsnaps};
+        READ_GENESIS_PARTIAL_FORESTINFO(h5_fd, "ForestInfoInFile", "ForestOffsetsAllSnaps", read_ndim, read_offset, read_count, gen->halo_offset_per_snap);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Variable 'macro_status' is assigned a value that is never used.](https://app.codacy.com/manual/jacobseiler/sage-model/pullRequest?prid=4838129)